src/main/java/uk/gov/companieshouse/acsp/manage/users/service/AcspMembersService.java

```diff
@@ -61,6 +62,45 @@ public List<AcspMembership> fetchAcspMemberships(final User user, final boolean
     return acspMembershipListMapper.daoToDto(acspMembers, user);
   }
 
+  @Transactional(readOnly = true)
+  public Optional<AcspMembersDao> fetchActiveAcspMember(final String userId) {
+    return acspMembersRepository.fetchActiveAcspMemberByUserId(userId);
+  }
+
+  @Transactional(readOnly = true)
+  public Optional<AcspMembersDao> fetchActiveAcspMemberByUserIdAndAcspNumber(
+      final String userId, final String acspNumber) {
+    return acspMembersRepository.fetchActiveAcspMemberByUserIdAndAcspNumber(userId, acspNumber);
+  }
+
+  public AcspMembersDao addAcspMember(
+      final String userId,
+      final String acspNumber,
+      final AcspMembership.UserRoleEnum userRole,
+      final String addedByUserId) {
+    final var now = LocalDateTime.now();
+    var newMembership = new AcspMembersDao();
+    newMembership.setUserId(userId);
+    newMembership.setAcspNumber(acspNumber);
+    newMembership.setUserRole(userRole);
+    newMembership.setCreatedAt(now);
+    newMembership.setAddedAt(now);
+    newMembership.setAddedBy(addedByUserId);
+    newMembership.setRemovedBy(null);
+    newMembership.setRemovedAt(null);
+    newMembership.setEtag(generateEtag());
+    return acspMembersRepository.insert(newMembership);
```

Comment:
Hmm - I don’t think we want to upsert (save), but just insert.  The LLD did say upset but I think through informal slack messages we decided against it.

Reply:
https://companieshouse.slack.com/archives/C067CGF6023/p1719995061780839 FYI

Reply:
- This endpoint is used to create an active association between a user and an ACSP.
- The PATCH endpoint can be used to disassociate the user and ACSP i.e. by updating the removed_by field.

Suppose that this endpoint added is used to add Elon Musk to Tesla, and then the PATCH endpoint is later used to remove Elon Musk from Tesla. Because this endpoint currently does not support update operations, there is no way to add Elon Musk back to Tesla. It would have to be an upsert or an insert and update.

Reply:
Kartheek said:

no update, keep POST simple.
POST : if no active membership, create one.
PATCH: if there is one and now removed, then remove.
POST again, check if its active, if not create new
[11:11](https://companieshouse.slack.com/archives/C067CGF6023/p1720433486792599)
no need to update.
once removed its removed.
11:11
dont change the state back to active from removed, dont let patch do that.